app/src/main/java/fr/free/nrw/commons/nearby/PlaceRenderer.java

```diff
@@ -103,27 +103,54 @@ protected void setUpView(View view) {
 
     @Override
     protected void hookListeners(View view) {
+        if (onBookmarkClick != null) {
+            final View.OnClickListener listener = view12 -> {
+                Timber.d("Renderer clicked");
+                TransitionManager.beginDelayedTransition(buttonLayout);
+
+                if (buttonLayout.isShown()) {
+                    closeLayout(buttonLayout);
+                } else {
+                    openLayout(buttonLayout);
+                }
 
-        final View.OnClickListener listener = view12 -> {
-            Timber.d("Renderer clicked");
-            TransitionManager.beginDelayedTransition(buttonLayout);
+            };
+            view.setOnClickListener(listener);
 
-            if (buttonLayout.isShown()) {
-                closeLayout(buttonLayout);
-            } else {
-                openLayout(buttonLayout);
-            }
+            view.requestFocus();
+            view.setOnFocusChangeListener((view1, hasFocus) -> {
+                if (!hasFocus && buttonLayout.isShown()) {
+                    closeLayout(buttonLayout);
+                } else if (hasFocus && !buttonLayout.isShown()) {
+                    listener.onClick(view1);
+                }
+            });
+        }else {
+            final View.OnClickListener listener = view12 -> {
+                Timber.d("Renderer clicked");
+                TransitionManager.beginDelayedTransition(buttonLayout);
+
+                if (buttonLayout.isShown()) {
+                    closeLayout(buttonLayout);
+                } else {
+                    openLayout(buttonLayout);
+                }
+                ((NearbyMapFragment)((NearbyFragment)((NearbyListFragment)fragment).getParentFragment()).getChildFragmentManager().findFragmentByTag(NearbyMapFragment.class.getSimpleName())).setMapCenter(place);
 
-        };
-        view.setOnClickListener(listener);
-        view.requestFocus();
-        view.setOnFocusChangeListener((view1, hasFocus) -> {
-            if (!hasFocus && buttonLayout.isShown()) {
-                closeLayout(buttonLayout);
-            } else if (hasFocus && !buttonLayout.isShown()) {
-                listener.onClick(view1);
-            }
-        });
+
+
+            };
+            view.setOnClickListener(listener);
+
+            view.requestFocus();
+            view.setOnFocusChangeListener((view1, hasFocus) -> {
+                if (!hasFocus && buttonLayout.isShown()) {
+                    closeLayout(buttonLayout);
+                } else if (hasFocus && !buttonLayout.isShown()) {
+                    listener.onClick(view1);
+                }
+            });
+        }
```

Comment:
The code between 143 - 153 also duplicated. Please optimize this part of code too.